modules/app/controllers/webhook.js

```diff
@@ -377,53 +379,7 @@ exports.updateWebhook = (req, res) => {
           })
         break
       case 'charge.succeeded':
-        return models.Order.update(
-          {
-            paid: paid,
-            status: status
-          },
-          {
-            where: {
-              source_id: event.data.object.source.id,
-              source: event.data.object.id
-            },
-            returning: true
-          }
-        )
-          .then(order => {
-            if (order[0]) {
-              return models.User.findOne({
-                where: {
-                  id: order[1][0].dataValues.userId
-                }
-              })
-                .then(user => {
-                  if (user) {
-                    if (paid && status === 'succeeded') {
-                      const language = user.language || 'en'
-                      i18n.setLocale(language)
-                      SendMail.success(
-                        user.dataValues,
-                        i18n.__('mail.webhook.payment.update.subject'),
-                        i18n.__('mail.webhook.payment.approved.message', {
-                          amount: event.data.object.amount / 100
-                        })
-                      )
-                    }
-                  }
-
-                  return res.json(req.body)
-                })
-                .catch(e => {
-                  return res.status(400).send(e)
-                })
-            }
-          })
-          .catch(e => {
-            return res.status(400).send(e)
-          })
-
-        break
+        return chargeSucceeded(event, paid, status, req, res)
```

Comment:
@VictorAlessander That's an amazing refactoring 🎉 

We can use this solution to split all webhook calls in modules 👍